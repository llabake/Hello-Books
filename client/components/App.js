import React, { Component } from 'react';
import { render } from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import requireAuthentication from './Authentication';
import HomePage from './HomePage';
import SignUpForm from './SignUpForm';
import SignInForm from './SignInForm';
import AllBooks from './main/AllBooks'
import AddBook from './adminDashBoard/AddBook';
import SingleBook from './main/SingleBook';
import AdminDashBoard from './adminDashBoard/AdminDashBoard';
import UserProfilePage from './userDashBoard/UserProfilePage';
import Favorite from './userFavorite/Favorite';
import SearchResult from './common/SearchResult';
import EditProfile from './userDashBoard/EditProfile';
import NotFound from './NotFound';

import '../styles/style.scss';
import Header from './common/Header';
import Footer from './common/Footer';
/**
 * 
 * 
 * @export
 * @class App
 * @extends {Component}
 */
export default class App extends Component {

  componentDidMount() {
    $('.button-collapse').sideNav({
      menuWidth: 250, // Default is 300
      edge: 'left', // Choose the horizontal origin
      draggable: true // Choose whether you can drag to open on touch screens
    }
  );
  // $('.carousel').carousel({
  //   dist: 0, // Perspective zoom. 
  //   shift: 0, // Set the spacing of the center item.
  //   padding: 50, // Set the padding between non center items.
  //   indicators: true, // Show indicators. Default is false
  // });
  // autoplay()
  // function autoplay() {
    // $('.carousel').carousel('next');
    // setTimeout(autoplay, 4500)}

    $('.slider').slider();
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    $('.tooltipped').tooltip({ delay: 50 });
    $('.tabs').tabs();
    $('.tooltipped').tooltip();
  

    // $('.modal').modal();
    // $('select').material_select();
    
  }
  /**
   * 
   * 
   * @returns {object} main App
   * @memberof App
   */
  render () {
    return (
      <Router>
        <div className="body">
        <header>
        <Header />
        </header>
        <main>
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route exact path='/signup' component={SignUpForm} />
          <Route exact path='/signin' component={SignInForm} />
          <Route exact path='/allbooks' component={requireAuthentication(AllBooks)} />
          <Route exact path='/addbook' component={requireAuthentication(AddBook)} />
          <Route exact path='/admindashboard' component={requireAuthentication(AdminDashBoard)} />
          <Route exact path='/book/:bookId' component={requireAuthentication(SingleBook)} />
          <Route exact path='/profile' component={requireAuthentication(UserProfilePage)} />
          <Route exact path='/favorite' component={requireAuthentication(Favorite)}/>
          <Route exact path='/search/:searchTerm' component={requireAuthentication(SearchResult)}/>
          <Route exact path='/editprofile' component={requireAuthentication(EditProfile)} />
          <Route component={NotFound} />
        </Switch>
        </main>
        <footer>
        <Footer />
        </footer>
        </div>
      </Router>
    );
  }
}





