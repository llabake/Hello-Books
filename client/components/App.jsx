import React, { Component } from 'react';
import { render } from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import requireAuthentication from './Authentication';
import HomePage from './HomePage';
import SignUpForm from './SignUpForm';
import SignInForm from './SignInForm';
import AllBooks from './main/AllBooks'
import AddBook from './adminDashBoard/AddBook';
import SingleBook from './main/SingleBook';
import AdminDashBoard from './adminDashBoard/AdminDashBoard';
import UserProfilePage from './userDashBoard/UserProfilePage';
import Favorite from './userFavorite/Favorite';
import SearchResult from './common/SearchResult';
import EditProfile from './userDashBoard/EditProfile';
import NotFound from './NotFound';

import '../styles/style.scss';
import Header from './common/Header';
import Footer from './common/Footer';
/**
 * 
 * 
 * @export
 * @class App
 * @extends {Component}
 */
export default class App extends Component {
  /**
   *@returns {void}
   *
   * @memberof App
   */
  componentDidMount() {
    $('.button-collapse').sideNav({
      menuWidth: 250, // Default is 300
      edge: 'left', // Choose the horizontal origin
      draggable: true // Choose whether you can drag to open on touch screens
    }
    );
    $('.slider').slider({
      full_width: true,
    }
    );
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    $('.tooltipped').tooltip({ delay: 50 });
    $('ul.tabs').tabs();
    $('.tooltipped').tooltip({ delay: 50 });
  }

  /**
   * 
   * 
   * @returns {object} main App
   * @memberof App
   */
  render() {
    return (
      <Router>
        <div className="body">
          <header>
            <Header />
          </header>
          <main>
            <Switch>
              <Route exact path='/' component={HomePage} />
              <Route exact path='/signup' component={SignUpForm} />
              <Route exact path='/signin' component={SignInForm} />
              <Route exact path='/allbooks' component={requireAuthentication(AllBooks)} />
              <Route exact path='/addbook' component={requireAuthentication(AddBook)} />
              <Route exact path='/admindashboard' component={requireAuthentication(AdminDashBoard)} />
              <Route exact path='/book/:bookId' component={requireAuthentication(SingleBook)} />
              <Route exact path='/profile' component={requireAuthentication(UserProfilePage)} />
              <Route exact path='/favorite' component={requireAuthentication(Favorite)} />
              <Route exact path='/search/:searchTerm' component={requireAuthentication(SearchResult)} />
              <Route exact path='/editprofile' component={requireAuthentication(EditProfile)} />
              <Route component={NotFound} />
            </Switch>
          </main>
          <footer>
            <Footer />
          </footer>
        </div>
      </Router>
    );
  }
}





