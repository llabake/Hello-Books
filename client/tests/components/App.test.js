
import React from 'react'
import { createStore } from 'redux'
import { Provider } from 'react-redux'
import { MemoryRouter, Switch } from 'react-router-dom'

import reducer from '../../reducers'
import App from '../../components/App';

test('The App component mounts correctly', () => {
  let spy = jest.spyOn(App.prototype, 'componentDidMount');

  const store = createStore(reducer)
  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <Switch>
          <App />
        </Switch>
      </MemoryRouter>
    </Provider>)
  expect(wrapper).toMatchSnapshot();
  expect(spy).toHaveBeenCalled();


})