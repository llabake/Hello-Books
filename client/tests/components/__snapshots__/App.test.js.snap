// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App component mounts correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "5asdow",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "5asdow",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Switch>
        <App
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          location={
            Object {
              "hash": "",
              "key": "5asdow",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
        >
          <BrowserRouter>
            <Router
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "blank",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            >
              <div
                className="body"
              >
                <header>
                  <Connect(Header)>
                    <Header
                      authenticated={false}
                      logout={[Function]}
                      user={Object {}}
                    >
                      <header>
                        <div
                          className="navbar-fixed"
                        >
                          <nav>
                            <div
                              className="nav-wrapper"
                            >
                              <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                className="brand-logo left adjust"
                                to="/"
                              >
                                <Route
                                  path="/"
                                >
                                  <Link
                                    aria-current={false}
                                    className="brand-logo left adjust"
                                    replace={false}
                                    to="/"
                                  >
                                    <a
                                      aria-current={false}
                                      className="brand-logo left adjust"
                                      href="/"
                                      onClick={[Function]}
                                    >
                                      HelloBooks
                                    </a>
                                  </Link>
                                </Route>
                              </NavLink>
                              <a
                                className="button-collapse"
                                data-activates="mobile-demo"
                                href="#"
                              >
                                <i
                                  className="material-icons"
                                >
                                  menu
                                </i>
                              </a>
                              <ul
                                className="right hide-on-med-and-down"
                                id="nav-mobile"
                              >
                                <li>
                                  <Connect(SearchBar)>
                                    <SearchBar
                                      dispatch={[Function]}
                                      searchedBooks={Array []}
                                    >
                                      <form
                                        onSubmit={[Function]}
                                      >
                                        <div
                                          className="input-field col s6 s12 "
                                        >
                                          <input
                                            name="searchTerm"
                                            onChange={[Function]}
                                            placeholder="Title, Author, Keyword"
                                            type="search"
                                            value=""
                                          />
                                          <label
                                            className="label-icon icon-sit"
                                            htmlFor="search"
                                          >
                                            <i
                                              className="material-icons clickable"
                                              onClick={[Function]}
                                            >
                                              search
                                            </i>
                                          </label>
                                          <i
                                            className="material-icons"
                                            style={
                                              Object {
                                                "fontSize": "1.5rem",
                                                "marginTop": "-6px",
                                              }
                                            }
                                          >
                                            close
                                          </i>
                                        </div>
                                      </form>
                                    </SearchBar>
                                  </Connect(SearchBar)>
                                </li>
                                <li>
                                  <a
                                    className="dropdown-button"
                                    data-activates="dropdown1"
                                  >
                                    Hi, User
                                    <i
                                      className="material-icons large left"
                                    >
                                      account_circle
                                    </i>
                                    <i
                                      className="material-icons right"
                                    >
                                      arrow_drop_down
                                    </i>
                                  </a>
                                  <ul
                                    className="dropdown-content"
                                    id="dropdown1"
                                  >
                                    <li>
                                      <Link
                                        replace={false}
                                        to="/signup"
                                      >
                                        <a
                                          href="/signup"
                                          onClick={[Function]}
                                        >
                                          Sign Up
                                        </a>
                                      </Link>
                                    </li>
                                    <li>
                                      <Link
                                        replace={false}
                                        to="/signin"
                                      >
                                        <a
                                          href="/signin"
                                          onClick={[Function]}
                                        >
                                          Sign In
                                        </a>
                                      </Link>
                                    </li>
                                    <li
                                      className="divider"
                                    />
                                  </ul>
                                </li>
                              </ul>
                            </div>
                          </nav>
                        </div>
                        <SideNav
                          authenticated={false}
                          handleLogout={[Function]}
                          user={Object {}}
                        >
                          <ul
                            className="side-nav show-on-medium-and-down"
                            id="mobile-demo"
                          >
                            <li>
                              <Connect(SearchBar)>
                                <SearchBar
                                  dispatch={[Function]}
                                  searchedBooks={Array []}
                                >
                                  <form
                                    onSubmit={[Function]}
                                  >
                                    <div
                                      className="input-field col s6 s12 "
                                    >
                                      <input
                                        name="searchTerm"
                                        onChange={[Function]}
                                        placeholder="Title, Author, Keyword"
                                        type="search"
                                        value=""
                                      />
                                      <label
                                        className="label-icon icon-sit"
                                        htmlFor="search"
                                      >
                                        <i
                                          className="material-icons clickable"
                                          onClick={[Function]}
                                        >
                                          search
                                        </i>
                                      </label>
                                      <i
                                        className="material-icons"
                                        style={
                                          Object {
                                            "fontSize": "1.5rem",
                                            "marginTop": "-6px",
                                          }
                                        }
                                      >
                                        close
                                      </i>
                                    </div>
                                  </form>
                                </SearchBar>
                              </Connect(SearchBar)>
                            </li>
                            <li
                              className="no-padding"
                            >
                              <ul
                                className="collapsible collapsible-accordion"
                              >
                                <li>
                                  <a
                                    className="collapsible-header"
                                  >
                                    Hi, User
                                    <i
                                      className="material-icons"
                                    >
                                      account_circle
                                    </i>
                                    <i
                                      className="material-icons right"
                                    >
                                      arrow_drop_down
                                    </i>
                                  </a>
                                  <div
                                    className="collapsible-body"
                                  >
                                    <ul>
                                      <li>
                                        <Link
                                          replace={false}
                                          to="/signup"
                                        >
                                          <a
                                            href="/signup"
                                            onClick={[Function]}
                                          >
                                            Sign Up
                                          </a>
                                        </Link>
                                      </li>
                                      <li>
                                        <Link
                                          replace={false}
                                          to="/signin"
                                        >
                                          <a
                                            href="/signin"
                                            onClick={[Function]}
                                          >
                                            Sign In
                                          </a>
                                        </Link>
                                      </li>
                                    </ul>
                                  </div>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </SideNav>
                      </header>
                    </Header>
                  </Connect(Header)>
                </header>
                <main>
                  <Switch>
                    <Route
                      component={[Function]}
                      computedMatch={
                        Object {
                          "isExact": false,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "blank",
                          "search": "",
                          "state": undefined,
                        }
                      }
                    >
                      <NotFound
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "createHref": [Function],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "pathname": "blank",
                              "search": "",
                              "state": undefined,
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        location={
                          Object {
                            "hash": "",
                            "pathname": "blank",
                            "search": "",
                            "state": undefined,
                          }
                        }
                        match={
                          Object {
                            "isExact": false,
                            "params": Object {},
                            "path": "/",
                            "url": "/",
                          }
                        }
                      >
                        <div
                          className="container"
                        >
                          <div
                            id="not-found"
                            style={
                              Object {
                                "marginLeft": "0px",
                              }
                            }
                          >
                            <h4>
                              This is embarassing, something's missing.
                            </h4>
                            <b>
                              404: page not found
                            </b>
                          </div>
                          <div>
                            <Connect(SearchBar)>
                              <SearchBar
                                dispatch={[Function]}
                                searchedBooks={Array []}
                              >
                                <form
                                  onSubmit={[Function]}
                                >
                                  <div
                                    className="input-field col s6 s12 "
                                  >
                                    <input
                                      name="searchTerm"
                                      onChange={[Function]}
                                      placeholder="Title, Author, Keyword"
                                      type="search"
                                      value=""
                                    />
                                    <label
                                      className="label-icon icon-sit"
                                      htmlFor="search"
                                    >
                                      <i
                                        className="material-icons clickable"
                                        onClick={[Function]}
                                      >
                                        search
                                      </i>
                                    </label>
                                    <i
                                      className="material-icons"
                                      style={
                                        Object {
                                          "fontSize": "1.5rem",
                                          "marginTop": "-6px",
                                        }
                                      }
                                    >
                                      close
                                    </i>
                                  </div>
                                </form>
                              </SearchBar>
                            </Connect(SearchBar)>
                          </div>
                          <div>
                            <ul
                              style={
                                Object {
                                  "textAlign": "center",
                                }
                              }
                            >
                              <li
                                style={
                                  Object {
                                    "display": "block",
                                  }
                                }
                              >
                                <Link
                                  replace={false}
                                  to="/"
                                >
                                  <a
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    Home
                                  </a>
                                </Link>
                              </li>
                              <li
                                style={
                                  Object {
                                    "display": "block",
                                  }
                                }
                              >
                                <Link
                                  replace={false}
                                  to="/allbooks"
                                >
                                  <a
                                    href="/allbooks"
                                    onClick={[Function]}
                                  >
                                    Books
                                  </a>
                                </Link>
                              </li>
                              <li
                                style={
                                  Object {
                                    "display": "block",
                                  }
                                }
                              >
                                <Link
                                  replace={false}
                                  to="/profile"
                                >
                                  <a
                                    href="/profile"
                                    onClick={[Function]}
                                  >
                                    Profile
                                  </a>
                                </Link>
                              </li>
                            </ul>
                          </div>
                        </div>
                      </NotFound>
                    </Route>
                  </Switch>
                </main>
                <footer>
                  <Footer>
                    <footer
                      className="page-footer center"
                    >
                      <div
                        className="container"
                      >
                        <div
                          className="row"
                        >
                          <div
                            className="col s12 m4 l3"
                          >
                            <h5
                              className="white-text"
                            >
                              My Account
                            </h5>
                            <ul
                              className="navigation"
                            >
                              <li>
                                <Link
                                  className="grey-text text-lighten-4"
                                  replace={false}
                                  to="/"
                                >
                                  <a
                                    className="grey-text text-lighten-4"
                                    href="/"
                                    onClick={[Function]}
                                  >
                                    Home
                                  </a>
                                </Link>
                              </li>
                              <li>
                                <Link
                                  className="white-text text-lighten-4"
                                  replace={false}
                                  to="/profile"
                                >
                                  <a
                                    className="white-text text-lighten-4"
                                    href="/profile"
                                    onClick={[Function]}
                                  >
                                    Profile
                                  </a>
                                </Link>
                              </li>
                              <li>
                                <Link
                                  className="grey-text text-lighten-4"
                                  replace={false}
                                  to="/favorite"
                                >
                                  <a
                                    className="grey-text text-lighten-4"
                                    href="/favorite"
                                    onClick={[Function]}
                                  >
                                    Favorite
                                  </a>
                                </Link>
                              </li>
                            </ul>
                          </div>
                          <div
                            className="col s12 m4 l6"
                          >
                            <h5
                              className="white-text"
                            >
                              Hello Books
                            </h5>
                            <p
                              className="white-text text-lighten-4"
                            >
                              Stay Connected. Join our over 5 million users today
                            </p>
                          </div>
                          <div
                            className="col s12 m4 l3"
                          >
                            <h5
                              className="white-text"
                            >
                              Be Social
                            </h5>
                            <ul
                              className="list-inline"
                            >
                              <li>
                                <a
                                  className="social-btn"
                                  href="https://facebook.com/hellobooksapp"
                                >
                                  <i
                                    className="fa fa-facebook-square fa-2x icon-style"
                                  />
                                </a>
                              </li>
                              <li>
                                <a
                                  className="social-btn"
                                  href="https://instagram.com/hellobooksapp"
                                >
                                  <i
                                    className="fa fa-instagram fa-2x icon-style"
                                  />
                                </a>
                              </li>
                              <li>
                                <a
                                  className="social-btn"
                                  href="https://twitter.com/hellobooksapp"
                                >
                                  <i
                                    className="fa fa-twitter-square fa-2x icon-style"
                                  />
                                </a>
                              </li>
                              <li>
                                <a
                                  className="social-btn"
                                  href="https://pinterest.com/hellobooksapp"
                                >
                                  <i
                                    className="fa fa-pinterest-square fa-2x icon-style"
                                  />
                                </a>
                              </li>
                            </ul>
                          </div>
                        </div>
                      </div>
                      <div
                        className="footer-copyright center-align"
                      >
                        <div
                          className="container"
                        >
                          © 
                          2018
                           Happy Girl World
                        </div>
                      </div>
                    </footer>
                  </Footer>
                </footer>
              </div>
            </Router>
          </BrowserRouter>
        </App>
      </Switch>
    </Router>
  </MemoryRouter>
</Provider>
`;
